monai/inferers/utils.py

```diff
@@ -278,8 +278,8 @@ def sliding_window_inference(
                 else:
                     output_image_list[0][o_slice] += sw_device_buffer[0].to(device=device)
             else:
-                sw_device_buffer[ss] *= w_t
                 sw_device_buffer[ss] = sw_device_buffer[ss].to(device)
+                sw_device_buffer[ss] *= w_t
```

Comment:
this `sw_device_buffer[ss] *= w_t` should happen on `sw_device`... I'm curious how you run into issues, maybe somehow the `predictor` returned a `device` tensor by mistake? it should return an output on `sw_device`.